### DISCUSSION TITLE
Cannot build nor run example/demo script- likely a problem with dependencies linking.

### DISCUSSION THREAD

**Discussion by yassinetb** [ocornut/imgui#5232] [New Users Build/Link/Run issues ONLY!]
Hi, 

Apologies if this should have gone to issues rather than discussion, and for the basic aspect of the question. 

I am working on Windows10 and trying to get examples to run before building my project - following the recommendation of a friend, I am first looking at "example_glfw_opengl3". When building the file with "make" command on the specific _example_glfw_opengl3_ directory, I get the following error: 

> g++: error: unrecognized command-line option '--cflags'

I tried various modifications in the makefile, which eventually allowed to successfully build, but when running code, I always got a massive number of "main.cpp:(.text+0x45b): undefined reference to" errors, which included both ImGUI functions and glfw3 functions. If I understand correctly, this means that my modifications never allowed the makefile to do its job of linking dependencies properly. 

I also get "undefined reference" errors when trying to run imguidemo.cpp, which leads me to think I also have dependencies linking issues in general. I still haven't managed to find what the problem is after a good day of research.

I am unsure what to do and am considering simply setting up ubuntu on my machine instead of losing time to debug what seems to be Windows-specific struggles.

**Comment by ocornut** [comment]
Hello Jassine, 

Those wouldn’t be Windows-specific. Linking errors means you aren’t linking the compiled object file that contains that symbol. Check out the final gcc command line, the .o/.obj files should be all included in the linker command line. (In a first step the source files are compiled into object files, in the final link step all the object files are assembled into an executable).

This is more a question of how to use Make and general understanding of the C++ compiling and linking process. You’ll find plenty of intormation on using Makefiles. If you are on Windows i would personally advise to use Visual Studio.

**Reply by yassinetb** [reply]
Thanks for the prompt reply. I'll investigate what you suggested and update the status of the discussion accordingly.

**Reply by yassinetb** [reply]
Switching to Visual Studio essentially solved the issue as I was able to directly operate with the VC++ Project files with pre-established additional include directoris etc... Now I'm excited to start building :)

**Reply by ocornut** [reply]
In addition you now have easy access to a debugger that doesn’t suck.
